plugins/main/bridge/bridge.go

```diff
@@ -289,6 +293,14 @@ func ensureBridge(brName string, mtu int, promiscMode, vlanFiltering bool) (*net
 		return nil, err
 	}
 
+	// Remove native vlan in case the bridge already exists and has one set
+	if vlanFiltering && *br.VlanDefaultPVID != defaultPVID {
+		err = netlink.BridgeSetVlanDefaultPVID(br, defaultPVID)
```

Comment:
This is problematic - the bridge may be managed by something else (e.g. NetworkManager) and the default PVID may be set intentionally. I think that unless asked to, the CNI should only manipulate the device it creates (veth), not the bridge it connects to

Reply:
We can say the same for promiscuous mode which is above. It's also changed even when the bridge is present.   
 Also, the bridge might have been created by the cni itself from a different attachment.

Anyway, I agree with you, something doesn't add up. I was expecting the function "bridgeByName" to also check that the attributes are the same and return and error otherwise, but it only checks whether the device is a bridge or not. In my opinion, that should be the case to force external managed bridge to comply with the requested configuration and/or to prevent different attachments to change the bridge configuration. The first attachment would set the configuration.

What do you think? c.c. @maiqueb

Reply:
For transparency, we discussed offline if it would make sense to expose yet another knob allowing the user to configure the default PVID of the bridge. This would be a somewhat cumbersome way of preserving the PVID of the bridge - which may be valid in some use cases.

I didn't understand if that would happen on this PR, or on a follow-up @mlguerrero12 .

Reply:
I would argue that this "default-pvid" parameter should be only applied on ports. Since the CNI creates these ports, it will be safe to do. In this case, CNI's "default-pvid" attribute can be simply a boolean: true -> do nothing, false -> strip the default pvid from the port.

The problem with changing anything on the bridge itself is, that if the bridge is managed by something other than the CNI (e.g. NetworkManager), we may end up with two processes setting bridge's attributes up and down.

Reply:
Conceptually, I agree. 

But I think that if the user has a way to specify the default pvid via CNI conf, it possible to persist the PVID value of the bridge it doesn't own. 

... sure, that's cumbersome, involves extra-steps, and may be error prone (a user forgets to do this, for instance).

Having said that, I think this alternative is the simplest thing we can do to tackle the issue; we'll discuss this a bit more, see if we can come up w/ a better alternative though. 

Maybe disallowing CNI from setting args on the bridge when it finds the bridge **already exists** (i.e. it would be able to attach veths to it) would work ?... I still fear that could cause regressions elsewhere.

Reply:
will create another pr with a different approach (following @phoracek suggestion. It's what nmstate does if I'm not mistaken)